CharlieBackend.Data/Repositories/Impl/CourseRepository.cs

```diff
@@ -66,5 +66,12 @@ public async Task<List<MentorCoursesDto>> GetMentorCourses(long id)
                         Name = x.Name
                     }).ToListAsync();
         }
+
+        public async Task<bool> IsCourseActive(long id)
+        {
+            var course = await _applicationContext.Courses.AsNoTracking().FirstOrDefaultAsync(c => c.Id == id);
+
+            return course.IsActive;
```

Comment:
What if someone decides to use this method and passes the `Id` of a course that doesn't exist. And then line 74 will throw [`NullReferenceException`](https://docs.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-5.0).

Please correct me if I'm wrong.

Reply:
I use method IsCourseActive inside the UpdateCourseAsync. If someone put non-existent id in UpdateCourseAsync, we get NotFound Error, due to this condition:

`if (!await _unitOfWork.CourseRepository.IsEntityExistAsync(id))
                {
                    return Result<CourseDto>.GetError(ErrorCode.NotFound, "Course Not Found");
                }`

So IsCourseActive method will never get a non-existent id.

Reply:
I agree, in the given context of `UpdateCourseAsync()` you won't get an exception.

What I'm saying is that you're introducing a whole new method for `CourseRepository`, which can be used wherever. It means if someone decides to use it anywhere else (without previously checking whether course with a given `Id` exists or not) - it will throw an exception.

Reply:
Ok, I changed this logic. We will get false, if course with a given Id doesn't exist. https://github.com/ita-social-projects/WhatBackend/pull/387/commits/48809ecac443cf879cf7bca7ce231e15b0133f42